tac/governing-documents/project-lifecycle.md

```diff
@@ -142,3 +142,128 @@ state independent of its prior history.
+
+Below is a guide for evaluating each of these criteria, with break-ups into sub-criteria that can be directly evaluated using one of three methods:
+1. Checking their OpenSSF Scorecard scores
+2. Checking their LFX Insights Best Practice Scores (produced by CLOMonitor)
+3. Manual inspection of code repositories
+
+(*Note*: The term "close to" below implies that a high (or highest) score is not mandatory but highly desired. If a project falls short on some such criteria but is otherwise mature and well-functioning, the TAC may ignore those criteria while encouraging the project's maintainers to strive for a higher score.)
+
+### Legal Criteria
+
+- (OpenSSF Score): *License*: MUST be 10
+
+### Diversity Criteria
+
+- (OpenSSF Score): *Contributors*: SHOULD be close to 10
+- (Manual): the `MAINTAINERS.md` files in the project's repositories MUST collectively indicate the involvement of more than one organization
+
+- Evaluation:
+  * To be a *Graduated* project, the number of organizations from which maintainers are drawn must be at least 3, and the OpenSSF Score SHOULD be 10.
+  * To be an *Incubated* project, the number of organizations from which maintainers are drawn must be at least 2.
+
+### Releases
+
+- (OpenSSF Score): *Packaging*: SHOULD be 10
+- (Manual): the number and frequency of releases in the recent past (quarter/year), as identified by inspecting the project's repositories, must be reasonably high (at least for one of the repositories)
+
+### Testing and Q/A
+
+- (OpenSSF Score): *CI Tests*: SHOULD be 10
+
+### Security
+
+- (OpenSSF Score): *Dangerous Workflow*: MUST be 10
+- (OpenSSF Score): *Token Permissions*: MUST be 10
+- (OpenSSF Score): *Branch-Protection*: MUST be 9 (Tier 4)
+- (OpenSSF Score): *Dependency-Update-Tools*: MUST be 10
+- (OpenSSF Score): *Fuzzing*: SHOULD be close to 10
+- (OpenSSF Score): *Pinned-Dependencies*: SHOULD NOT be 0
+- (OpenSSF Score): *SAST*: SHOULD be close to 10
+- (OpenSSF Score): *Security-Policy*: SHOULD be at least 9
+- (OpenSSF Score): *Signed-Releases*: (*TBD* from Security Artifacts Task Force's conclusions) tentatively SHOULD be at least 8
+- (OpenSSF Score): *Token-Permissions*: MUST be 10
+- (OpenSSF Score): *Vulnerabilities*: SHOULD be 10
+- (Manual): the contents of the `SECURITY.md` files in the project's repositories must comply with [LFDT requirements](./security.md)
+
+### Structure
+
+- (LFX Insights Best Practice Score): SHOULD be 50% or higher (averaged over all repositories)
+- (Manual): check if the repositories follow the [repository structure guidelines](./repository-structure.md)
+
+### Maintenance
+
+- (OpenSSF Scorecard): *Code Review*: MUST be 10
+- (Manual): frequency of pull requests, Discord activity, and mailing list activity, must be reasonably high
+
+### Production
+
+- (Manual): an `ADOPTERS.md` file must be present in at least one of the project's repositories, and contain at least one reference to the project's real-world usage
```

Comment:
Adoption is outside the control of project maintainers, and changes to adoption (both adoption and abandonment) usually happens without maintainers knowing. Sometimes we do catch word of an adopter, but that doesn't mean it should be highlighted as a 'showcase' adopter by the project.

I guess I'm questioning the value of trying to maintain something that we don't usually know about and which changes frequently and which may be a misleading representation in the first place.

Reply:
When Tracy and I were discussing this in one of the task force meetings, we thought the presence of this file could just be a useful heuristic for the TAC (and users) to evaluate a project's maturity. Personally, I think having a record of even past usages of a project is useful, but we don't have to penalize a project for not having (or knowing about) any active usage.

But I take your point about this being outside the control of maintainers. How about we instruct projects to add a suggestion in the main `README.md` for future adopters to submit a PR to the `ADOPTERS.md` whenever they start actively using it (and another PR if/when they abandon it)? This way it'll be a community-driven effort. For sure, many adopters may not bother (especially if there is confidentiality involved), but some likely will.

Reply:
Agreed, makes sense to instruct users to PR their own additions to the adopters file.